pkg/server/plugin/nodeattestor/aws/iid_test.go

```diff
@@ -0,0 +1,424 @@
+	stream, err := p.Attest(context.Background())
+	defer stream.CloseSend()
+	resp, err := stream.Recv()
+	s.requireErrorContains(err, "not configured")
+	s.Require().Nil(resp)
+}
+
+func (s *IIDAttestorSuite) TestErrorOnEmptyRequest() {
+	_, err := s.attest(&nodeattestor.AttestRequest{})
+	s.requireErrorContains(err, "request missing attestation data")
+}
+
+func (s *IIDAttestorSuite) TestErrorOnInvalidType() {
+	_, err := s.attest(&nodeattestor.AttestRequest{
+		AttestationData: &nodeattestor.AttestationData{
+			Type: "foo",
+		},
+	})
+	s.requireErrorContains(err, `unexpected attestation data type "foo"`)
+}
+
+func (s *IIDAttestorSuite) TestErrorOnMissingData() {
+	data := &common.AttestationData{
+		Type: aws.PluginName,
+	}
+
+	_, err := s.attest(&nodeattestor.AttestRequest{AttestationData: data})
+	s.requireErrorContains(err, "unexpected end of JSON input")
+}
+
+func (s *IIDAttestorSuite) TestErrorOnBadData() {
+	data := &common.AttestationData{
+		Type: aws.PluginName,
+		Data: make([]byte, 0),
+	}
+
+	_, err := s.attest(&nodeattestor.AttestRequest{AttestationData: data})
+	s.requireErrorContains(err, "unexpected end of JSON input")
+}
+
+func (s *IIDAttestorSuite) TestErrorOnAlreadyAttested() {
+	data := &common.AttestationData{
+		Type: aws.PluginName,
+		Data: s.iidAttestationDataToBytes(*s.buildDefaultIIDAttestationData()),
+	}
+
+	_, err := s.attest(&nodeattestor.AttestRequest{
+		AttestationData: data,
+		AttestedBefore:  true,
+	})
+	s.requireErrorContains(err, "the IID has been used and is no longer valid")
+}
+
+func (s *IIDAttestorSuite) TestErrorOnBadSignature() {
+	iid := s.buildDefaultIIDAttestationData()
+	iid.Signature = "bad sig"
+	data := &common.AttestationData{
+		Type: aws.PluginName,
+		Data: s.iidAttestationDataToBytes(*iid),
+	}
+
+	_, err := s.attest(&nodeattestor.AttestRequest{
+		AttestationData: data,
+	})
+	s.requireErrorContains(err, "illegal base64 data at input byte")
+}
+
+func (s *IIDAttestorSuite) TestErrorOnNoSignature() {
+	iid := s.buildDefaultIIDAttestationData()
+	iid.Signature = ""
+	data := &common.AttestationData{
+		Type: aws.PluginName,
+		Data: s.iidAttestationDataToBytes(*iid),
+	}
+
+	_, err := s.attest(&nodeattestor.AttestRequest{
+		AttestationData: data,
+	})
+	s.requireErrorContains(err, "verifying the cryptographic signature")
+}
+
+// get a DescribeInstancesOutput with essential structs created, but no values
+// (device index and root device type) filled out
+func getDefaultDescribeInstancesOutput() ec2.DescribeInstancesOutput {
+	return ec2.DescribeInstancesOutput{
+		Reservations: []*ec2.Reservation{
+			&ec2.Reservation{
+				Instances: []*ec2.Instance{
+					&ec2.Instance{
+						NetworkInterfaces: []*ec2.InstanceNetworkInterface{
+							&ec2.InstanceNetworkInterface{
+								Attachment: &ec2.InstanceNetworkInterfaceAttachment{},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func (s *IIDAttestorSuite) TestClientAndIDReturns() {
```

Comment:
are we missing a case for the SkipBlockDevice and SkipEC2Calling configurations?